app/auth/models.py

```diff
@@ -152,3 +153,42 @@ def create_user(self, email, full_name, user_name, password):
             return response.json()
         else:
             return None
+
+
+class FileData(object):
+
+    def __init__(self, package_name, publisher,
+                 relative_path, props):
+        self.package_name = package_name
+        self.publisher = publisher
+        self.relative_path = relative_path
+        self.props = props
+        self.bitstore = BitStore(publisher=publisher,
+                                 package=package_name)
+
+    def _generate_bitstore_url(self):
+        kwargs = {}
+        if 'type' in self.props:
+            kwargs['file_type'] = self.props['type']
+        if 'acl' in self.props:
+            kwargs['acl'] = self.props['acl']
+        post = self.bitstore.\
+            generate_pre_signed_post_object(self.relative_path,
+                                            md5=self.props['md5'],
+                                            **kwargs)
+        return post
+
+    def get_response(self):
```

Comment:
Why is a model object got stuff about making a response for the API (I assume this is a response for the API??). That could indicate a poor separation of concerns.